server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -2001,16 +2003,17 @@ static RescoreVector fromIndexOptions(Map<String, ?> indexOptionsMap) {
             if (oversampleNode == null) {
                 throw new IllegalArgumentException("Invalid rescore_vector value. Missing required field " + OVERSAMPLE);
             }
-            return new RescoreVector((float) XContentMapValues.nodeDoubleValue(oversampleNode));
-        }
-
-        RescoreVector {
-            if (oversample < 1) {
+            float oversampleValue = (float) XContentMapValues.nodeDoubleValue(oversampleNode);
+            if (oversampleValue == 0 && indexVersion.before(RESCORE_PARAMS_ALLOW_ZERO_TO_QUANTIZED_VECTORS)) {
+                throw new IllegalArgumentException("oversample must be greater than 1");
+            }
+            if (oversampleValue < 1) {
```

Comment:
I feel that I'm missing something obvious here, but `0` is now a valid value for `oversampleValue` in new index versions, right?

Reply:
yeah, let me verify I am capturing this accurately in tests. I might have removed an `else if` accidentally.